src/order/prime_ideal.lean

```diff
@@ -85,10 +92,76 @@ lemma prime_pair.I_is_prime (IF : prime_pair P) : is_prime IF.I :=
 { compl_filter := by { rw IF.compl_I_eq_F, exact IF.F.is_pfilter },
   ..IF.I_is_proper }
 
+end preorder
+
+section semilattice_inf
+
+variable [semilattice_inf P]
+
+variables {x y : P} {I : ideal P}
+
+lemma is_prime.mem_or_mem (hI : is_prime I) {x y : P} : x ⊓ y ∈ I → x ∈ I ∨ y ∈ I :=
+begin
+  contrapose!,
+  let F := hI.compl_filter.to_pfilter,
+  show x ∈ F ∧ y ∈ F → x ⊓ y ∈ F,
+  exact λ h, inf_mem _ _ h.1 h.2,
+end
+
+lemma is_prime.of_mem_or_mem [is_proper I] (hI : ∀ {x y : P}, x ⊓ y ∈ I → x ∈ I ∨ y ∈ I) :
+  is_prime I :=
+begin
+  rw is_prime_iff,
+  use ‹_›,
+  apply is_pfilter.of_def,
+  { exact set.nonempty_compl.2 (I.is_proper_iff.1 ‹_›) },
+  { intros x _ y _,
+    refine ⟨x ⊓ y, _, inf_le_left, inf_le_right⟩,
+    have := mt hI,
+    tauto!,
+   },
+  { exact @compl_mem_of_ge _ _ _}
+end
+
+lemma is_prime_iff_mem_or_mem [is_proper I] : is_prime I ↔ ∀ {x y : P}, x ⊓ y ∈ I → x ∈ I ∨ y ∈ I :=
+⟨is_prime.mem_or_mem, is_prime.of_mem_or_mem⟩
+
+end semilattice_inf
+
+section distrib_lattice
+
+variables [distrib_lattice P]
+variables {I : ideal P}
+
+@[priority 100]
+instance is_maximal.is_prime [is_maximal I] : is_prime I :=
```

Comment:
For such a nice theorem, and the main point of this PR, it's almost too bad that the statement is so unassuming!